lib/command/run.rb

```diff
@@ -48,80 +63,178 @@ class Run < Base
       # Overrides remote CPLN_TOKEN env variable with local token.
       # Useful when superuser rights are needed in remote container.
       cpl run -a $APP_NAME --use-local-token -- bash
+
+      # Replaces the existing Dockerfile entrypoint with `/bin/sh -c`
+      # (default Dockerfile command when no entrypoint is specified).
+      cpl run -a $APP_NAME --entrypoint none -- rails db:migrate
+
+      # Replaces the existing Dockerfile entrypoint.
+      cpl run -a $APP_NAME --entrypoint /app/alternative-entrypoint.sh -- rails db:migrate
       ```
     EX
 
-    attr_reader :location, :workload_to_clone, :workload_clone, :container
+    INTERACTIVE_COMMANDS = [
+      "bash",
+      "rails console",
+      "rails c",
+      "rails dbconsole",
+      "rails db"
+    ].freeze
+
+    attr_reader :interactive, :detached, :location, :workload_to_clone, :workload_clone,
+                :container, :job, :replica, :command
 
     def call # rubocop:disable Metrics/MethodLength
+      @interactive = config.options[:interactive] || interactive_command?
+      @detached = config.options[:detached]
+
       @location = config.location
-      @workload_to_clone = config.options["workload"] || config[:one_off_workload]
-      @workload_clone = "#{workload_to_clone}-run-#{random_four_digits}"
+      @workload_to_clone = config.options[:workload] || config[:one_off_workload]
+      @workload_clone = "#{workload_to_clone}-run"
 
-      step("Cloning workload '#{workload_to_clone}' on app '#{config.options[:app]}' to '#{workload_clone}'") do
-        clone_workload
+      if cp.fetch_workload(workload_clone).nil?
+        step("Cloning workload '#{workload_to_clone}' to '#{workload_clone}'") do
+          clone_workload
+        end
+      end
+      step("Updating workload '#{workload_clone}' based on '#{workload_to_clone}'") do
+        update_workload_clone
+      end
+
+      unless interactive
+        # Catch Ctrl+C
+        trap("SIGINT") do
+          print_detached_commands
+          exit(ExitCodes::INTERRUPT)
+        end
       end
 
       wait_for_workload(workload_clone)
-      wait_for_replica(workload_clone, location)
-      run_in_replica
-    ensure
+
+      data = build_data
+      @job = start_cron_workload(workload_clone, data, location)
+      @replica = wait_for_replica_for_job(workload_clone, job, location)
+
       progress.puts
-      ensure_workload_deleted(workload_clone)
+      if interactive
+        run_interactive
+      else
+        run_non_interactive
+      end
+    ensure
+      if interactive
+        progress.puts
+        stop_replica(workload_clone, replica, location)
```

Comment:
It's important that we have a timeout for interactive commands in case the program quits.

Can we have the main job program be something other than `tail -f` so it can check if anybody is connected to it and start checking that after a few minutes, and if no other process is connected, then issue command to terminate itself.

Reply:
ask for help from control plane channel on how to do this.